tests/unittests/test_dev_mode.py

```diff
@@ -0,0 +1,74 @@
+import json
+import os
+import unittest
+from unittest.mock import patch, MagicMock
+
+import singer
+
+from tap_jira.http import Client
+
+
+LOGGER = singer.get_logger()
+
+
+class TestClientDevMode(unittest.TestCase):
+    """Test the dev mode functionality."""
+
+    def setUp(self):
+        """Creates a sample config for test execution"""
+        # Data to be written
+        self.mock_config = {
+            "oauth_client_secret": "sample_client_secret",
+            "user_agent": "test_user_agent",
+            "oauth_client_id": "sample_client_id",
+            "access_token": "sample_access_token",
+            "cloud_id": "1234567890",
+            "refresh_token": "sample_refresh_token",
+            "start_date": "2017-12-04T19:19:32Z",
+            "request_timeout": 300,
+            "groups": "jira-administrators, site-admins, jira-software-users",
+        }
+
+        self.tmp_config_filename = "sample_jira_config.json"
+
+        # Serializing json
+        json_object = json.dumps(self.mock_config, indent=4)
+        # Writing to sample_quickbooks_config.json
```

Comment:
Comment says `sample_quickbooks_config` instead of `sample_jira_config`

Reply:
Will fix it.